Content.Client/UserInterface/Systems/Radial/RadialUiController.cs

```diff
@@ -0,0 +1,139 @@
+
+namespace Content.Client.UserInterface.Systems.Radial;
+
+[UsedImplicitly]
+public sealed class RadialUiController : UIController
+{
+    [Dependency] private readonly IEntityManager _entityManager = default!;
+    [Dependency] private readonly IEyeManager _eyeManager = default!;
+    [Dependency] private readonly IPlayerManager _playerManager = default!;
+    [Dependency] private readonly IConfigurationManager _cfg = default!;
+
+    private LayoutContainer _radials = default!;
+    private List<RadialContainer> _attachedRadials = new();
+
+    public override void Initialize()
+    {
+        _radials = new LayoutContainer();
+
+        var gameplayStateLoad = UIManager.GetUIController<GameplayStateLoadController>();
+        gameplayStateLoad.OnScreenLoad += OnScreenLoad;
+        gameplayStateLoad.OnScreenUnload += OnScreenUnload;
+    }
+
+    public void OnScreenLoad()
+    {
+        _radials.DisposeAllChildren();
+        _attachedRadials.Clear();
+
+        var viewportContainer = UIManager.ActiveScreen!.FindControl<LayoutContainer>("ViewportContainer");
+        SetRadialsRoot(viewportContainer);
+    }
+
+    public void OnScreenUnload()
+    {
+        _radials.DisposeAllChildren();
+        _attachedRadials.Clear();
+    }
+
+    public override void FrameUpdate(FrameEventArgs args)
+    {
+        base.FrameUpdate(args);
+
+        foreach (var radial in _attachedRadials.Cast<RadialContainer>())
+        {
+            if (_entityManager.Deleted(radial.AttachedEntity))
+            {
+                radial.Dispose();
+                return;
+            }
+
+            if (!_entityManager.TryGetComponent<TransformComponent>(radial.AttachedEntity, out var xform) || xform.MapID != _eyeManager.CurrentMap)
+            {
+                return;
+            }
+
+            var localPlayer = _playerManager.LocalPlayer;
+            if (localPlayer == null)
+                return;
+
+            // Check distance beetween entities
+            if (_entityManager.TryGetComponent<TransformComponent>(localPlayer.ControlledEntity, out var myxform))
+            {
+                var onePoint = xform.WorldPosition;
+                var twoPoint = myxform.WorldPosition;
+                var distance = (onePoint - twoPoint).Length();
+
+                if (radial.VisionRadius < distance)
+                {
+                    radial.Dispose();
+                    return;
+                }
+            }
+
+            var offset = (-_eyeManager.CurrentEye.Rotation).ToWorldVec() * -radial.VerticalOffset;
+            var worldPos = xform.WorldPosition + offset;
+
+            var lowerCenter = _eyeManager.WorldToScreen(worldPos) / _radials.UIScale;
+            var screenPos = lowerCenter - new Vector2(radial.DesiredSize.X / 2, radial.DesiredSize.Y / 2);
+            // Round to nearest 0.5
+            screenPos = (screenPos * 2).Rounded() / 2;
+            LayoutContainer.SetPosition(radial, screenPos);
+        }
+    }
```

Comment:
You shouldn't assume a control is related to the current eye as this breaks with overlays.

Reply:
But SpeechBubble.cs works same. What i can do?